MDBF 653 - Run Connect Tests as part of one builder in buildbot

RazvanLiviuVarzaru opened the pull request:
As part of MDBF-653 we need to run Connect storage engine tests on a Builder.

Testing:
 [amd64-ubuntu-2004-fulltest/18](https://buildbot.dev.mariadb.org/#/builders/183/builds/18)
- see [connect tests](https://buildbot.dev.mariadb.org/#/builders/183/builds/18/steps/27/logs/stdio)
- other tests were compared to a previous run [amd64-ubuntu-2004-fulltest/15 ](https://buildbot.dev.mariadb.org/#/builders/183/builds/15)

Selective regression testing:
- addTests function was modified to be general purpose so I've tested an autogen builder and compared it with a previous run to confirm that tests ran the same way as before.
  - Test:  [aarch64-almalinux-8/16](https://buildbot.dev.mariadb.org/#/builders/124/builds/16)

vladbogo commented:
@RazvanLiviuVarzaru looks good, but I think it would be better to have two PRs and two tasks: one for connect and one for refactoring. Other than that, I couldn't spot any mistake, but the refactoring changes should affect almost all builders so I would say it's better to separate them

RazvanLiviuVarzaru commented:
> @RazvanLiviuVarzaru looks good, but I think it would be better to have two PRs and two tasks: one for connect and one for refactoring. Other than that, I couldn't spot any mistake, but the refactoring changes should affect almost all builders so I would say it's better to separate them

@vladbogo The only way around I see at this moment to separate them is to use add connect tests to the function that I've deleted in this PR "add_test_steps".
Indeed, many builders are affected by addTests refactoring as they use quickbuild factory. I tried to make sure that there is no impact by:
- keeping the positional arguments as they were before
- providing default values for keyword arguments that will go in MTR list of parameters

The only reason I've modified the addTests function here is that it's preparing the ground for further adoption by other masters.
